src/features/clusters/currentClusterSlice.ts

```diff
@@ -33,6 +34,13 @@ export const currentClusterSlice = createSlice({
   name: 'currentCluster',
   reducers: {
     updateCluster: (state, action: PayloadAction<Cluster>) => ({ ...state, data: action.payload }),
+    updateHost: (state, action: PayloadAction<Host>) => {
+      const hostIndex = _.findIndex(state.data?.hosts, (host) => host.id === action.payload.id);
+      if (hostIndex >= 0) {
+        _.set(state, `data.hosts[${hostIndex}]`, action.payload);
```

Comment:
you should not modify state directly

Reply:
redux toolkit uses Immer behind the scenes which allows using direct mutations in reducers. https://redux-toolkit.js.org/api/createReducer#direct-state-mutation

I agree that it is not ideal to overuse this type of magic but in this case it makes the code quite simpler.